docs/modules/ROOT/pages/how-to-guides/proc_prefetching-dependencies-to-support-hermetic-build.adoc

```diff
@@ -0,0 +1,195 @@
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip`
+Cachi2 supports pip by parsing *requirements.txt* files placed in the root of your repository and then downloading the specified dependencies. These files should be lockfiles, which include all the transitive dependencies. Each transitive dependency in the *requirements.txt* file must be pinned to a specific version.
+
+.Prerequisites
+* You have an upgraded build pipeline.
+* You have dependencies outside your git repository.
+
+.Procedures
+For each component that you want to build hermetically:
+
+. Navigate to your component’s source code.
+. Open the *.tekton* directory and locate the *.yaml* files related to the *pull-request* and *push* processes.
+. *Set up dependency prefetching with Cachi2:* In both the `.yaml` files, add the following prefetch-input pipeline parameter.
+
++
+[source,yaml]
+----
+param:
+    -  name: prefetch-input
+       value: <package-manager-lock-file> <1>
+----
+<1> If the package manager lock file is located in the root directory of your project, enter `pip`. Otherwise, if the package manager lock file is in a subdirectory, provide the path to the lock file in the JSON format. For example,
+{"type": "{<package-manager>}", "path": "path-to-the-directory-which-contains-the-lockfile}"}
+
+. Create a file (for example, pyproject.toml) with metadata (name and version) and place it in the root of your repository.
+
++
+[source,metadata]
+----
+[metadata]
+name = "my_package"
+version = "0.1.0"
+----
+
++
+NOTE: Instead of a *pyproject.toml* file, you can also create a *setup.py* or *setup.cfg* file. For information on the metadata of these files, see link:https://github.com/containerbuildsystem/cachi2/blob/main/docs/pip.md#project-metadata[Project Metadata].
+
+. To generate a fully resolved requirements.txt file, run
+
++
+[source,command]
+----
+$ pip-compile pyproject.toml --generate-hashes
+----
+
++
+NOTE: The command mentioned assumes that you have defined project dependencies in *pyproject.toml*. However, if you have defined the project dependencies in either the *setup.py* or *setup.cfg*, ensure to update the command accordingly.
+
+include::../partials/con_hermetic_verification.adoc[]
+
+== Enabling prefetch builds for `pip` with source dependencies
+If your project depends on `pip` with source dependencies, you can prefetch builds to improve the performance of the build process. Prefetch builds allow Cachi2 to fetch the build dependencies and provide them to Pip for offline installation in a network-isolated environment.
+
+.Prerequisites
+
+* You have an upgraded build pipeline.
+* You have dependencies outside of your git repository.
+* Download the link:https://raw.githubusercontent.com/containerbuildsystem/cachito/master/bin/pip_find_builddeps.py[`pip_find_builddeps.py`] script directly from GitHub.
```

Comment:
What does this script achieve? Shouldn't this live below in the procedures as it automates part of the work of identifying the build dependencies?

Reply:
Thank you!

@chmeliik @brunoapimentel 
1. It's listed as a prerequisite here, do you think we can move it in the procedures? https://github.com/containerbuildsystem/cachi2/blob/806f868947382aef0ddf4a14fd7f3996274bfcfc/docs/pip.md#requirements-buildtxt

2. What is the location we need to save this file after downloading it?

Reply:
> It's listed as a prerequisite here, do you think we can move it in the procedures?

yes :slightly_smiling_face: 

Downloading the script is a prerequisite to generating requirements-build.txt, but the whole thing is part of the procedure that prepares a repo

> What is the location we need to save this file after downloading it?

* anywhere in $PATH and make it executable
* or anywhere at all and run it as `python path/to/pip_find_builddeps.py <args>`

Reply:
I think it makes sense to download the script as part of the procedure as it is optional to the overall process and only required to some specific part of the process.